content/en/blog/general/introducing-kyverno-envoy-plugin/index.md

```diff
@@ -0,0 +1,236 @@
+Starting HTTP server on Port 8000
+Starting GRPC server on Port 9000
+Request is initialized in kyvernojson engine .
+2024/04/26 17:11:42 Request passed the deny-guest-request-at-post policy rule.
+Request is initialized in kyvernojson engine .
+2024/04/26 17:22:11 Request violation: -> POST method calls at path /book are not allowed to guests users
+ -> any[0].check.request.http.headers.authorization.(split(@, ' ')[1]).(jwt_decode(@ , 'secret').payload.role): Invalid value: "guest": Expected value: "admin"
+-> GET method call is allowed to both guest and admin users
+ -> any[1].check.request.http.headers.authorization.(split(@, ' ')[1]).(jwt_decode(@ , 'secret').payload.role): Invalid value: "guest": Expected value: "admin"
+ -> any[1].check.request.http.method: Invalid value: "POST": Expected value: "GET"
+-> GET method call is allowed to both guest and admin users
+ -> any[2].check.request.http.method: Invalid value: "POST": Expected value: "GET"
+Request is initialized in kyvernojson engine .
+2024/04/26 17:23:13 Request passed the deny-guest-request-at-post policy rule.
+Request is initialized in kyvernojson engine .
+2024/04/26 17:23:55 Request passed the deny-guest-request-at-post policy rule.
+```
+
+### Configuration
+
+To deploy Kyverno-Envoy-Plugin include the following container in your Kubernetes Deployments:
+
+```yaml
+- name: kyverno-envoy-plugin
+  image: sanskardevops/plugin:0.0.34
+  imagePullPolicy: IfNotPresent
+  ports:
+    - containerPort: 8181
+    - containerPort: 9000
+  volumeMounts:
+    - readOnly: true
+  args:
+    - "serve"
+    - "--policy=/policies/policy.yaml"
+    - "--address=:9000"
+    - "--healthaddress=:8181"
+  livenessProbe:
+    httpGet:
+      path: /health
+      scheme: HTTP
+      port: 8181
+    initialDelaySeconds: 5
+    periodSeconds: 5
+  readinessProbe:
+    httpGet:
+      path: /health
+      scheme: HTTP
+      port: 8181
+    initialDelaySeconds: 5
+    periodSeconds: 5  
+```
+
+## Conclusion
+
+This blog post demonstrated how the Kyverno-Envoy-Plugin can be effectively used to make external authorization decisions for incoming requests in a microservice architecture. By leveraging the power of Kyverno policies and Envoy's External Authorization filter, you can achieve fine-grained, context-aware access control without modifying your application code. This approach not only simplifies the management of security policies but also enhances the security posture of your services by ensuring that authorization checks are consistently applied across your microservices.
+
+The Kyverno-Envoy-Plugin provides a robust solution for organizations looking to enforce policy-driven access controls in their service meshes. By following the steps outlined in this guide, you can easily deploy and configure the plugin, allowing you to take full advantage of Kyverno's policy capabilities in conjunction with Envoy's powerful proxy features.
```

Comment:
Kyverno-Envoy-Plugin should all three words be hyphenated? Maybe Kyverno-Envoy plugin

Reply:
I used the hyphenated form "Kyverno-Envoy-Plugin" because the repository name follows that convention. I'm open to changing it to "Kyverno-Envoy plugin" if that would be more appropriate.